index.d.ts

```diff
@@ -0,0 +1,35 @@
+import { HttpsAgent } from 'agentkeepalive';
+import { AgentOptions, Agent } from 'http';
+
+interface Logger {
+    info: () => void;
+    warn: () => void;
+    error: () => void;
+}
+
+export interface AgentPoolOptions {
+    agentType: Agent;
+    maxAgents: number;
+    destroyTime: number;
+    logger?: Logger;
+}
+    
+export interface KeepAliveAgentOptions extends AgentOptions {
+    keepAliveTimeout: number;
```

Comment:
is it used anywhere?

Reply:
No. But it makes the `KeepAliveAgentOptions ` as backward compatible which is not mentioned in the API of `agentkeepalive` docs

Reply:
I would extend it to https://github.com/node-modules/agentkeepalive/blob/master/index.d.ts#L23 and then add that param as well as optional

Reply:
Removed for now then